bgpd/bgp_route.c

```diff
@@ -845,6 +871,13 @@ static int bgp_path_info_cmp(struct bgp *bgp, struct bgp_path_info *new,
 		return 0;
 	}
 
+	/* Here if these are imported routes then get ultimate pi for path
+	 * compare. */
+	new = bgp_get_bpi_ultimate(new);
```

Comment:
How do you know if _these_ are imported?

Reply:
```c
/*
 * Get the ultimate path info.
 */
static struct bgp_path_info* bgp_get_bpi_ultimate(struct bgp_path_info *info)
{
	struct bgp_path_info *bpi_ultimate;

	if(info->sub_type != BGP_ROUTE_IMPORTED)
		return info;
```

Reply:
Okay, but it a bit confuses a reader when changed in the middle of the processing. Can't it be processed at the beginning since you already return the original path info if not IMPORTED?

Reply:
These changes are removed from this commit. Since it requires topotest changes. It will be committed later.